Lib/pathlib/_os.py

```diff
@@ -92,11 +93,31 @@ def copyfd(source_fd, target_fd):
 
 
 if _winapi and hasattr(_winapi, 'CopyFile2'):
-    def copyfile(source, target):
+    def is_dirlink(path):
```

Comment:
I imagine this will get used again sooner rather than later - probably better to use `if os.name == 'nt'` as the check, rather than relying on `_winapi` being present (or check that `st_file_attributes` and `st_reparse_tag` exist, but that's a bit more complicated).

Strictly speaking, it's possible for `_winapi` to be present or absent independent of whether we're on Windows, whereas `os.name` represents the behaviour of the `os` module directly (which is what we're relying on here) and the stat members are what we actually expect to exist, so either of those checks would be better.